spatialdata/_core/transform.py

```diff
@@ -1,86 +1,526 @@
+            ndim = len(translation)
+
+        assert type(translation) == np.ndarray
+        assert len(translation) == ndim
+        self.translation = translation
+        self._ndim = ndim
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "translation",
+            "translation": self.translation.tolist(),
+        }
+
+    def transform_points(self, points: ArrayLike) -> ArrayLike:
+        return points + self.translation
+
+    def inverse(self) -> BaseTransformation:
+        return Translation(translation=-self.translation)
+
+    def to_affine(self) -> Affine:
+        assert self.ndim == 2
+        m: ArrayLike = np.vstack(
+            (np.hstack((np.array([[1.0, 0.0], [0.0, 1.0]]), self.translation.reshape(2, 1))), np.array([0.0, 0.0, 1.0]))
+        )
+        return Affine(affine=m)
+
+
+class Scale(BaseTransformation):
+    def __init__(
+        self,
+        scale: Optional[Union[ArrayLike, List[Any]]] = None,
+        ndim: Optional[int] = None,
+    ) -> None:
+        """
+        class for storing scale transformations.
+        """
+        if scale is None:
+            assert ndim is not None
+            scale = np.ones(ndim, dtype=float)
+
+        if ndim is None:
+            assert scale is not None
+            if isinstance(scale, list):
+                scale = np.array(scale)
+            ndim = len(scale)
+
+        assert type(scale) == np.ndarray
+        assert len(scale) == ndim
+        self.scale = scale
+        self._ndim = ndim
+
+    @property
+    def ndim(self) -> Optional[int]:
+        return self._ndim
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            "type": "scale",
+            "scale": self.scale.tolist(),
+        }
+
+    def transform_points(self, points: ArrayLike) -> ArrayLike:
+        return points * self.scale
+
+    def inverse(self) -> BaseTransformation:
+        new_scale = np.zeros_like(self.scale)
+        new_scale[np.nonzero(self.scale)] = 1 / self.scale[np.nonzero(self.scale)]
+        return Scale(scale=new_scale)
+
+    def to_affine(self) -> Affine:
+        assert self.ndim == 2
+        m: ArrayLike = np.vstack((np.hstack((np.diag(self.scale), np.zeros((2, 1)))), np.array([0.0, 0.0, 1.0])))
+        return Affine(affine=m)
 
-        if scale_factors is None:
-            if translation is not None:
-                scale_factors = np.ones(len(translation), dtype=float)
-            elif ndim is not None:
-                scale_factors = np.ones(ndim, dtype=float)
+
+class Affine(BaseTransformation):
+    def __init__(
+        self,
+        affine: Optional[Union[ArrayLike, List[Any]]] = None,
+    ) -> None:
+        """
+        class for storing scale transformations.
+        """
+        if affine is None:
+            affine = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]], dtype=float)
```

Comment:
why not using `np.diag` ?

Reply:
or better `np.eye` or `np.identity` ?

Reply:
copilot typed it for me 😁 for the ndim case I'll use `np.eye`